pkg/api/deploymentapi/depresourceapi/new_deployment_test.go

```diff
@@ -463,16 +529,107 @@ func TestNew(t *testing.T) {
 				}},
 			}},
 		},
+		{
+			name: "Succeeds to create a deployment payload with ES, Kibana and EnterpriseSearch instances",
+			args: args{params: NewParams{
+				Version: "7.6.1",
+				Region:  "ece-region",
+				ElasticsearchInstance: InstanceParams{
+					RefID:     "main-elasticsearch",
+					Size:      1024,
+					ZoneCount: 1,
+				},
+				KibanaInstance: InstanceParams{
+					RefID:     "main-kibana",
+					Size:      1024,
+					ZoneCount: 1,
+				},
+				EnterpriseSearchInstance: InstanceParams{
+					RefID:     "main-enterprise_search",
+					Size:      1024,
+					ZoneCount: 1,
+				},
+				DeploymentTemplateID:   "default",
+				EnterpriseSearchEnable: true,
+				API: api.NewMock(
+					mock.New200Response(mock.NewStructBody(enterpriseSearchKibanaTemplateResponse)),
+					mock.New200Response(mock.NewStructBody(enterpriseSearchKibanaTemplateResponse)),
+					mock.New200Response(mock.NewStructBody(enterpriseSearchKibanaTemplateResponse)),
```

Comment:
Only 1 API call is required now
```suggestion
					mock.New200Response(mock.NewStructBody(enterpriseSearchKibanaTemplateResponse)),
```